components/script/serviceworker_manager.rs

```diff
@@ -155,20 +143,40 @@ impl ServiceWorkerManager {
                 if self.active_workers.contains_key(&scope) {
                     let _ = self.active_workers.remove(&scope);
                 } else {
-                    warn!("ScopeThings for {:?} is not active", scope);
+                    warn!("ServiceWorker for {:?} is not active", scope);
+                }
+                true
+            },
+            ServiceWorkerMsg::ForwardDOMMessage(msg, scope_url) => {
+                if self.active_workers.contains_key(&scope_url) {
```

Comment:
I had tried that, but the borrow checker complains:
```
error[E0500]: closure requires unique access to `self` but `self.active_workers` is already borrowed
   --> /home/creativcoder/dev/servo/components/script/serviceworker_manager.rs:145:70
    |
145 |                 let sender = self.active_workers.get(&scope).or_else(|| self.wakeup_serviceworker(scope).as_ref());
    |                              -------------------                     ^^ ---- borrow occurs due to use of `self` in closure
    |                              |                                       |
    |                              |                                       closure construction occurs here
    |                              borrow occurs here
...
152 |             }
    |             - borrow ends here

```
and that is why i took this verbose if else approach so that the outer `contains_key` check keeps the borrows on seperate branches. Suggestions ?